marlowe-playground-client/src/MainFrame/State.purs

```diff
@@ -647,13 +656,14 @@ handleGistAction settings PublishGist = do
 
     -- playground is a meta-data file that we currently just use as a tag to check if a gist is a marlowe playground gist
     playground = "{}"
-  marlowe <- pruneEmpty <$> toSimulation Simulation.editorGetValue
-  haskell <- pruneEmpty <$> HaskellEditor.editorGetValue
-  blockly <- pruneEmpty <$> query _blocklySlot unit (H.request Blockly.GetWorkspace)
-  javascript <- pruneEmpty <$> (toJavascriptEditor JavascriptEditor.editorGetValue)
-  actus <- pruneEmpty <$> query _actusBlocklySlot unit (H.request ActusBlockly.GetWorkspace)
+  workflow <- use _workflow
+  marlowe <- if workflow /= Just Marlowe then pure Nothing else pruneEmpty <$> MarloweEditor.editorGetValue
+  blockly <- if workflow /= Just Blockly then pure Nothing else pruneEmpty <$> BlocklyEditor.editorGetValue
+  haskell <- if workflow /= Just Haskell then pure Nothing else pruneEmpty <$> HaskellEditor.editorGetValue
+  javascript <- if workflow /= Just Javascript then pure Nothing else pruneEmpty <$> (toJavascriptEditor JavascriptEditor.editorGetValue)
+  actus <- if workflow /= Just Actus then pure Nothing else pruneEmpty <$> query _actusBlocklySlot unit (H.request ActusBlockly.GetWorkspace)
```

Comment:
This looks like it should be a `case` then, because only one of the `if`s is going to work. Since `files` is a record, it would be very concise if written using update syntax over a record that only has `Nothing`s on it (`mempty`?).

Reply:
Also I think the parenthesis in 663 is unnecessary (even though it was already there)

Reply:
nice, much clearer that there is only one option now.